pkg/controller/test/fake/secretlister.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fake
```

Comment:
I'm not a fan of `pkg/controller/test/fake` - it sounds like a fake test 😅

I don't think many places will need to make use of this sort of thing - perhaps the struct itself can be made un-exported and packing internal to where it is used?

Reply:
Or maybe `test/unit/listers`?